src/wgpu_engine.rs

```diff
@@ -25,13 +25,45 @@ pub struct WgpuEngine {
     pool: ResourcePool,
     bind_map: BindMap,
     downloads: HashMap<Id, Buffer>,
+    pub(crate) use_cpu: bool,
 }
 
-struct Shader {
+struct WgpuShader {
     pipeline: ComputePipeline,
     bind_group_layout: BindGroupLayout,
+}
+
+pub enum CpuShaderType {
+    Present(fn(u32, &[CpuBinding])),
+    Missing,
```

Comment:
I'm wondering why we need two different ways (`Missing` and `Unused`) to disable a CPU shader. I think we could just have two modes instead: a CPU shader is either present or it's not. Then at execution time we either use the GPU version or we use the CPU one. I think for now, this can be represented by two variants rather than three. WDYT?

Reply:
`Unused` exists because neither the CPU or GPU shader is needed when passing `use-cpu`, so we can skip compiling the GPU shader even if the CPU shader is missing.

Reply:
I see, `Unused` represents a stage that gets skipped when running in CPU mode. I think this only makes sense in the current global "run everything on the CPU setting" but once we support a combination of CPU and GPU pipelines, we will likely want to have the pipelines available regardless.

I think what I would prefer is to have a way to defer pipeline creation for an unused pipeline rather than skip it using a global setting (similarly to what we have with buffer resources and "proxies" in our engine abstraction). But that should happen as future work.